lxd/instance/drivers/driver_qemu_templates.go

```diff
@@ -1,46 +1,97 @@
 package drivers
 
 import (
+	"fmt"
+	"strings"
 	"text/template"
 )
 
-// Base config. This is common for all VMs and has no variables in it.
-var qemuBase = template.Must(template.New("qemuBase").Parse(`
-# Machine
-[machine]
-graphics = "off"
-{{if eq .architecture "x86_64" -}}
-type = "q35"
-{{end -}}
-{{if eq .architecture "aarch64" -}}
-type = "virt"
-gic-version = "max"
-{{end -}}
-{{if eq .architecture "ppc64le" -}}
-type = "pseries"
-cap-large-decr = "off"
-{{end -}}
-{{if eq .architecture "s390x" -}}
-type = "s390-ccw-virtio"
-{{end -}}
-accel = "kvm"
-usb = "off"
-
-{{if eq .architecture "x86_64" -}}
-[global]
-driver = "ICH9-LPC"
-property = "disable_s3"
-value = "1"
-
-[global]
-driver = "ICH9-LPC"
-property = "disable_s4"
-value = "1"
-{{end -}}
-
-[boot-opts]
-strict = "on"
-`))
+type cfgEntry struct {
+	key   string
+	value string
+}
+
+type cfgSection struct {
+	name    string
+	comment string
+	entries []cfgEntry
+}
+
+func qemuAppendSections(sb *strings.Builder, sections []cfgSection) {
```

Comment:
One usability idea here is to make the sections argument varidic so its easier to use one or multiple sections